### PR TITLE
Add option to append query and mutation to operation names

### PR DESCRIPTION
**Pull request by BenSchwab**
closes https://github.com/apollographql/apollo-android/issues/551

Also noticed that we had no tests for mutations in apollo compiler, so added one, and change the name of test files from `TestQuery.***` to `TestOperation.***`.

### CODE REVIEW

File: apollo-gradle-plugin/src/main/java/com/apollographql/apollo/gradle/ApolloExtension.java [hunk 1]
```diff
@@ -12,6 +12,7 @@
   private Map<String, String> customTypeMapping = new LinkedHashMap<>();
   private String nullableValueType = NullableValueType.ANNOTATED.getValue();
   private boolean generateAccessors = true;
+  private boolean useSemanticNaming = false;
```
Comment (BenSchwab) [inline apollo-gradle-plugin/src/main/java/com/apollographql/apollo/gradle/ApolloExtension.java]: This makes `useSemanticNaming` default to false. Wouldn't mind  making this true, if we want to default to same behavior as iOS.

File: apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/OperationTypeSpecBuilder.kt [hunk 2]
```diff
@@ -1,17 +1,29 @@
...
 
 class OperationTypeSpecBuilder(
     val operation: Operation,
-    val fragments: List<Fragment>
+    val fragments: List<Fragment>,
+    useSemanticNaming: Boolean
 ) : CodeGenerator {
-  private val OPERATION_TYPE_NAME = operation.operationName.capitalize()
-  private val DATA_VAR_TYPE = ClassName.get("", "$OPERATION_TYPE_NAME.Data")
+  private val OPERATION_TYPE_NAME: String
```
Comment (sav007) [inline apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/OperationTypeSpecBuilder.kt]: nit: not sure why we make these names to be capitalized initially, should be lower camel case I guess

**Comment by sav007** [comment]
What about fragment, should we do the same for them, just for consistency.

File: apollo-gradle-plugin/src/main/java/com/apollographql/apollo/gradle/ApolloExtension.java (hunk 1 above)
Comment (sav007) [inline apollo-gradle-plugin/src/main/java/com/apollographql/apollo/gradle/ApolloExtension.java]: don't have any objections, I wish we could put this on vote :)

**Comment by BenSchwab** [comment]
sure, I'll bump to 0.40. I decided to default to the same generated code as iOS - so it is a breaking change. However, by setting `useSemanticNaming = false`, it would not be a breaking change.

**Comment by BenSchwab** [comment]
Also regarding not including fragments -- iOS doesn't append "Fragment" to the end of fragments. I think this is the way to go because we currently reference fragments through a `fragments` list, so the code reads clearly. For example: 
```
fragment comparisonFields on Character {
```
would be accessed something like `data.fragments().comparisonFields()` which makes sense.

**Pull request merged**